src/Proxies/LegacyProxy.php

```diff
@@ -0,0 +1,92 @@
+<?php
+/**
+ * LegacyProxy class file.
+ *
+ * @package Automattic/WooCommerce/Tools/Proxies
+ */
+
+namespace Automattic\WooCommerce\Proxies;
+
+use \Psr\Container\ContainerInterface as Container;
+
+/**
+ * Proxy class to access legacy WooCommerce functionality.
+ *
+ * This class should be used to interact with code outside the `src` directory, especially functions and classes
+ * in the `includes` directory, unless a more specific proxy exists for the functionality at hand (e.g. `ActionsProxy`).
+ * Idempotent functions can be executed directly.
+ *
+ * @package Automattic\WooCommerce\Tools\Proxies
+ */
+class LegacyProxy {
+
+	/**
+	 * Gets an instance of a given legacy class.
+	 * This must not be used to get instances of classes in the `src` directory.
+	 *
+	 * @param string $class_name The name of the class to get an instance for.
+	 * @param mixed  ...$args Parameters to be passed to the class constructor or to the appropriate internal 'get_instance_of_' method.
+	 *
+	 * @return object The instance of the class.
+	 * @throws \Exception The requested class belongs to the `src` directory, or there was an error creating an instance of the class.
+	 */
+	public function get_instance_of( string $class_name, ...$args ) {
```

Comment:
this is not powerful enough, we also load certain classes like DataStore with `load` method instead of `instance` with an argument. perhaps we should also support a passing method name as well?

Reply:
The idea is that this method will grow over time with any special cases we need, so the first time we need to deal with classes implementing `load` methods we'll add the logic handling that. It's impossible to cover all possible cases so I just wanted to provide the most basic ones. That said, this is something worth adding to the documentation.